test/runnable/test711.d

```diff
@@ -0,0 +1,55 @@
+/*
+TEST_OUTPUT:
+---
+---
+*/
+string result;
+
+template Mixer()
+{
+    override void test()
+    {
+        result ~= "A";
+    }
+}
+
+class Foo
+{
+    void test()
+    {
+        result ~= "B";
+    }
+}
+
+class Bar : Foo
+{
+    mixin Mixer!() mixer;
+    override void test()
+    {
+        result ~= "C";
+        mixer.test();
+    }
+}
+
+class Bar2 : Foo
+{
+    override void test()
+    {
+        result ~= "C";
+        mixer.test();
+    }
+    mixin Mixer!() mixer;
+}
```

Comment:
Is the point of `Bar` and `Bar2` to show that it doesn't matter if the mixin is placed before or after `test`?

Reply:
Yes. Before this patch, if the mixin was before `test` it would have resulted in an infinite recursion. If the mixin would have been placed after `test` it would have resulted in an ICE.